pkg/scanners/clairify/clairify.go

```diff
@@ -411,12 +421,38 @@ func (c *clairify) GetVulnerabilities(image *storage.Image, components *clairGRP
 	return convertImageToImageScan(image.GetMetadata(), resp.GetImage()), nil
 }
 
+func callGRPCWithRetryOnError(ctx context.Context, operation string, f func() error) error {
+	err := wait.ExponentialBackoffWithContext(ctx, scannerDefaultBackoff, func() (bool, error) {
```

Comment:
Certainly, we're Kubernetes product and integrating with `k8s.io/apimachinery` is not that bad.

However, all current usages of `k8s.io/apimachinery/pkg/util/wait` are in functionality that I believe has problems running test on GHA and needs to be revised.

However, that's not my main point. We already have https://github.com/stackrox/stackrox/blob/46a27ca4f17b15756ee6a8becb78e0da4139497e/pkg/retry/retry.go#L33-L33 and already use `github.com/cenkalti/backoff/v3`. Would either of those fit the use? It is better for maintainability is to have fewer patterns. In this case, I suggest not broaden the usage of `k8s.io/apimachinery/pkg/util/wait` and use either of two other more established patterns.

Reply:
@msugakov:

>  all current usages of k8s.io/apimachinery/pkg/util/wait are in functionality that I believe has problems running test on GHA and needs to be revised.

Thanks. I was hoping to get pointers on that. I should have mentioned it in the description to make it clear, though.

> We already have pkg/retry/retry.go and use github.com/cenkalti/backoff/v3. Would either of those fit the use? 

I noticed `pkg/retry` when writing this, but [stackrox/pkg/retry/retry.go](https://github.com/stackrox/stackrox/blob/46a27ca4f17b15756ee6a8becb78e0da4139497e/pkg/retry/retry.go#L33-L33) doesn't support out-of-the-box jitter, has hardcoded retry intervals (100ms) and rate (x2), and more importantly can't tie the retries with the timeout context. And I intended to retry within the timeout limit. I did not find references to `cenkalti/backoff/v3` in that package.

I feel you suggest I could onboard cenkalti's `backoff.Backoff()` to `pkg/retry`, which I think is a good idea. But please, kindly confirm if that's indeed your original suggestion.